spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -192,30 +175,90 @@ public void onInit() {
 					this.rabbitProperties.getSsl().getTrustStorePassword());
 		}
 	}
 
 	@Override
 	public RabbitConsumerProperties getExtendedConsumerProperties(String channelName) {
-		return extendedBindingProperties.getExtendedConsumerProperties(channelName);
+		return this.extendedBindingProperties.getExtendedConsumerProperties(channelName);
 	}
 
 
 	@Override
 	public RabbitProducerProperties getExtendedProducerProperties(String channelName) {
-		return extendedBindingProperties.getExtendedProducerProperties(channelName);
+		return this.extendedBindingProperties.getExtendedProducerProperties(channelName);
 	}
 
 	@Override
-	public Binding<MessageChannel> doBindConsumer(String name, String group, MessageChannel inputChannel,
+	protected MessageProducer createConsumerEndpoint(String name, String group, Object queue,
 			ExtendedConsumerProperties<RabbitConsumerProperties> properties) {
-		boolean anonymousConsumer = !StringUtils.hasText(group);
-		String baseQueueName = anonymousConsumer ? groupedName(name, ANONYMOUS_GROUP_NAME_GENERATOR.generateName())
+
+		DirectChannel convertingBridgeChannel = new DirectChannel();
+		convertingBridgeChannel.setBeanFactory(this.getBeanFactory());
+		final String baseQueueName = baseQueueName(name, group);
+		convertingBridgeChannel.setBeanName(baseQueueName + ".bridge");
+
+		SimpleMessageListenerContainer listenerContainer = new SimpleMessageListenerContainer(
+				this.connectionFactory);
+		listenerContainer.setAcknowledgeMode(properties.getExtension().getAcknowledgeMode());
+		listenerContainer.setChannelTransacted(properties.getExtension().isTransacted());
+		listenerContainer.setDefaultRequeueRejected(properties.getExtension().isRequeueRejected());
+		int concurrency = properties.getConcurrency();
+		concurrency = concurrency > 0 ? concurrency : 1;
+		listenerContainer.setConcurrentConsumers(concurrency);
+		int maxConcurrency = properties.getExtension().getMaxConcurrency();
+		if (maxConcurrency > concurrency) {
+			listenerContainer.setMaxConcurrentConsumers(maxConcurrency);
+		}
+		listenerContainer.setPrefetchCount(properties.getExtension().getPrefetch());
+		listenerContainer.setRecoveryInterval(properties.getExtension().getRecoveryInterval());
+		listenerContainer.setTxSize(properties.getExtension().getTxSize());
+		listenerContainer.setTaskExecutor(new SimpleAsyncTaskExecutor(((Queue) queue).getName() + "-"));
+		listenerContainer.setQueues((Queue) queue);
+		int maxAttempts = properties.getMaxAttempts();
+		if (maxAttempts > 1 || properties.getExtension().isRepublishToDlq()) {
+			RetryOperationsInterceptor retryInterceptor = RetryInterceptorBuilder.stateless()
+					.maxAttempts(maxAttempts)
+					.backOffOptions(properties.getBackOffInitialInterval(),
+							properties.getBackOffMultiplier(),
+							properties.getBackOffMaxInterval())
+					.recoverer(determineRecoverer(baseQueueName, properties.getExtension().getPrefix(),
+							properties.getExtension().isRepublishToDlq()))
+					.build();
```

Comment:
Perhaps (in AB), we could extract the body...

	if (maxAttempts > 1) {
		buildRetryTemplate(properties);
 	}

And then call `buildRetryTemplate(properties)` from here, since we always need a retry template even in maxAttempts is 1 (and we want to republish errors).